### PR TITLE
Add global grid width, height and thickness settings

### PR DESCRIPTION
**Pull request by Miepee**
## Description
This PR adds global room grid width, height and thickness settings, which override the automatically calculated grid width / height and hardcoded grid thickness.

### Caveats
None

### Notes
This is Change 2 of #931.

### CODE REVIEW

**Comment by VladiStep** [comment]
The room grid settings should be separated, as it done with a profile mode settings.
Also, the settings labels are not centered with the text boxes.

**Comment by Miepee** [comment]
Addressed those

**Comment by VladiStep** [comment]
You didn't properly aligned the check boxes and their labels - you should add `VerticalAlignment="Center"` to each checkbox and label.

File: UndertaleModTool/Windows/SettingsWindow.xaml [hunk 1]
```diff
@@ -41,36 +41,47 @@
...
             <TextBlock Grid.Row="4" Grid.Column="0" Margin="25 2 2 2" Text="Automatically associate .win files" ToolTip="Automatic file association. Enabled by default."/>
-            <CheckBox Grid.Row="4" Grid.Column="2" Margin="3" Content="" IsChecked="{Binding TempRunMessageShow}"/>
+            <CheckBox Grid.Row="4" Grid.Column="2" Margin="3" Content=""  IsChecked="{Binding TempRunMessageShow}"/>
             <TextBlock Grid.Row="4" Grid.Column="2" Margin="25 2 2 2" Text="Warn about temp running" ToolTip="Warn about temp running. Enabled by default."/>
 
-            <CheckBox Grid.Row="5" Grid.Column="0" Margin="3" Content="" IsChecked="{Binding Warn_About_GMS23}"/>
+            <CheckBox Grid.Row="5" Grid.Column="0" Margin="3"  Content="" IsChecked="{Binding Warn_About_GMS23}"/>
             <TextBlock Grid.Row="5" Grid.Column="0" Margin="25 2 2 2" Text="Warn about GMS 2.3" ToolTip="Warn about GMS 2.3. Enabled by default."/>
-            <CheckBox Grid.Row="5" Grid.Column="2" Margin="3" Content="" IsChecked="{Binding UseGMLCache}"/>
+            <CheckBox Grid.Row="5" Grid.Column="2" Margin="3"  Content="" IsChecked="{Binding UseGMLCache}"/>
```
Comment (VladiStep) [inline UndertaleModTool/Windows/SettingsWindow.xaml:56]: You forgot to remove redundant spaces.

File: UndertaleModTool/Controls/UndertaleRoomRenderer.xaml.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public UndertaleRoomRenderer()
 
         private void RoomRenderer_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
         {
-            (DataContext as UndertaleRoom)?.SetupRoom(!bgGridDisabled);
+            UndertaleRoomEditor.SetupRoomWithGrids(DataContext as UndertaleRoom);
             UndertaleRoomEditor.GenerateSpriteCache(DataContext as UndertaleRoom);
```
Comment (VladiStep) [inline UndertaleModTool/Controls/UndertaleRoomRenderer.xaml.cs:57]:
```suggestion
            if (DataContext is not UndertaleRoom room)
                return;
            
            UndertaleRoomEditor.SetupRoomWithGrids(room);
            UndertaleRoomEditor.GenerateSpriteCache(room);
```

**Review by VladiStep — CHANGES_REQUESTED** [review]

File: UndertaleModLib/Models/UndertaleRoom.cs [hunk 3]
```diff
@@ -376,54 +377,50 @@ public void SetupRoom(bool calculateGrid = true)
...
+        {
+            Point scale = new((int) tile.Width, (int) tile.Height);
+            if (tileSizes.ContainsKey(scale))
+                tileSizes[scale]++;
+            else
+                tileSizes.Add(scale, 1);
+        }
+
+        // If tiles exist at all, grab the most used tile size and use that as our grid size
+        if (tileSizes.Count <= 0) return;
```
Comment (VladiStep) [inline UndertaleModLib/Models/UndertaleRoom.cs:417]: Add default size assignment (`16`) besides `return`.

File: UndertaleModLib/Models/UndertaleRoom.cs [hunk 4]
```diff
@@ -376,54 +377,55 @@ public void SetupRoom(bool calculateGrid = true)
...
+        foreach (Tile tile in tileList)
+        {
+            Point scale = new((int) tile.Width, (int) tile.Height);
+            if (tileSizes.ContainsKey(scale))
+                tileSizes[scale]++;
+            else
+                tileSizes.Add(scale, 1);
+        }
+
+        // If tiles exist at all, grab the most used tile size and use that as our grid size
```
Comment (VladiStep) [inline UndertaleModLib/Models/UndertaleRoom.cs:416]: Move this comment right before line 424.

**Review by VladiStep — APPROVED** [review]

**Pull request merged**